src/libvirt-dbus.js

```diff
@@ -1462,6 +1469,28 @@ function initResource(connectionName, method, updateOrAddMethod, flags) {
             .catch(ex => console.warn('initResource action failed:', ex.toString()));
 }
 
+export function migrateToUri(connectionName, objPath, destUri, storage, live, temporary) {
+    // direct migration is not supported by QEMU, so it's opposite, the P2P migration should always be used
+    let flags = Enum.VIR_MIGRATE_PEER2PEER;
+
+    if (!live) {
+        flags = flags | Enum.VIR_MIGRATE_OFFLINE | Enum.VIR_MIGRATE_PERSIST_DEST;
+    } else {
+        flags = flags | Enum.VIR_MIGRATE_LIVE;
+
+        if (!temporary)
+            flags = flags | Enum.VIR_MIGRATE_PERSIST_DEST;
+    }
+
+    if (storage === "copy")
+        flags = flags | Enum.VIR_MIGRATE_NON_SHARED_DISK;
+
+    if (!temporary)
+        flags = flags | Enum.VIR_MIGRATE_UNDEFINE_SOURCE;
+
+    return call(connectionName, objPath, 'org.libvirt.Domain', 'MigrateToURI3', [destUri, {}, flags], { timeout, type: 'sa{sv}u' });
```

Comment:
The 30 seconds timeout here is really too optimistic. (We should not have timeout for the migation API).

Reply:
Fixed.